### PR TITLE
lightning: fix in-correctly check for session variable

### PR DESCRIPTION
**Pull request by Little-Wallace**
### What problem does this PR solve?

Problem Summary:
I'm working for use tidb-backend to replace dm/loader. And I made a mistake in https://github.com/pingcap/tidb/pull/28471, where I set variables without checking them.

### What is changed and how it works?

I have tested it in https://github.com/pingcap/ticdc/pull/3220/files

### Check List

Tests 

- [ ] Integration test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by glorv — APPROVED** [review]

File: br/pkg/lightning/restore/tidb.go [hunk 1]
```diff
@@ -118,13 +118,20 @@ func DBFromConfig(ctx context.Context, dsn config.DBStore) (*sql.DB, error) {
 		}
 	}
 
+	if dsn.Vars != nil {
```
Comment (glorv) [inline br/pkg/lightning/restore/tidb.go:121]: Please remove L115-119

**Comment by glorv** [comment]
/hold 
before comment is resolved

**Review by lance6716 — APPROVED** [review]

**Pull request merged**